### ISSUE TITLE
in-memory file is much larger than in-disk file

### ISSUE THREAD

**Issue by twmht** [scikit-image/scikit-image#2583]
## Description
Trying to store image to in-memory file, but got much larger size

0.jpg
![image](https://cloud.githubusercontent.com/assets/1567200/24284611/133d4984-10a7-11e7-961e-90bcf0687a24.png)

## Way to reproduce
[If reporting a bug, please include the following important information:]

```
from skimage import io
from io import BytesIO
import os


b = io.imread('0.jpg')

# 46101 bytes
io.imsave('1.jpg', b)
real_size = int(os.path.getsize('1.jpg'))
print (real_size)

c = io.imread('1.jpg')

mem = BytesIO()
io.imsave(mem, c)
mem.seek(0)
byte = mem.read()
# 365451 bytes (much larger than in-disk file)
print (len(byte))
```

what I expected is that in-memory file is equal to in-disk file, but in-memory file is much larger than in-disk file.

Maybe the cause is that the image compression (like jpeg) do not apply to in-memory file, but I do not know how to specify the file format to `io.imsave` when trying to save to in-memory file.

Any advise?

thank you

**Comment by jni** [comment]
Hi @twmht, I believe you might need to fiddle with the `quality` keyword argument to `imsave` for JPEG images. Try replacing `io.imsave(mem, c)` with `io.imsave(mem, c, quality=q)` for various values of `q`. If this resolves the issue, please close it, otherwise, let us know with more updates. btw, I don't know if it's github's fault with how they handle attachments, but your file appears to me as a png, not jpg, so I couldn't reliably experiment.

**Comment by jni** [comment]
btw, sorry that it is not currently documented as an argument — that is a [work in progress](https://github.com/scikit-image/scikit-image/pull/2063).

**Comment by twmht** [comment]
@jni 

Here is the jpg.

![0](https://cloud.githubusercontent.com/assets/1567200/24501723/54ae8c36-157d-11e7-8f31-7c57901bfab7.jpg)

I used PIL and it works as I expected.

```
from io import BytesIO
import os
from PIL import Image


b = Image.open('0.jpg')

b.save('1.jpg')
real_size = int(os.path.getsize('1.jpg'))
# 46101 bytes
print (real_size)

c = BytesIO()
b.save(c, 'JPEG')
c.seek(0)
byte = c.read()
print (len(byte))
```
however, scikit-image used PIL as plugin, but I don't know how to specify the format when saving to in-memory file.

If scikit-image does not support that, I can submit a PR to that.

**Comment by jni** [comment]
@twmht here we go:

```python
In [30]: io.imsave(mem, image, format_str='JPEG', quality=75)

In [31]: mem.seek(0)
Out[31]: 0

In [32]: imgbytes = mem.read()

In [33]: len(imgbytes)
Out[33]: 46101
```

Notice the `format_str` and `quality` keyword arguments.

If you want to submit a PR to document this, it would be very welcome!

**Comment by jni** [comment]
@twmht I've merged #2063 which documents the "quality" argument. A pull request documenting `format_str` would also be welcome!

**Comment by twmht** [comment]
@jni 

thanks!
